### PR TITLE
bootctl: Print version number of detected binaries again

### PR DESCRIPTION
**Pull request by arianvp**
In 9b9f6aa41204c995b7a3b3e99c50e92b0a8b3dde (#27069) we changed get_file_version to return 0 instead of 1 on success. However we did not update the printing of the version number to handle this change so we would never print the version number.

This might need backport to v253 given the original PR was also back ported.

### CODE REVIEW

File: src/boot/bootctl-status.c [hunk 1]
```diff
@@ -243,10 +243,7 @@ static int enumerate_binaries(
                 /* Do not output this entry immediately, but store what should be printed in a state
                  * variable, because we only will know the tree glyph to print (branch or final edge) once we
                  * read one more entry */
-                if (r > 0)
-                        r = asprintf(previous, "/%s/%s (%s%s%s)", path, de->d_name, ansi_highlight(), v, ansi_normal());
-                else
-                        r = asprintf(previous, "/%s/%s", path, de->d_name);
```
Comment (arianvp) [inline src/boot/bootctl-status.c:249]: do we want to check for `-ESRCH` here and print anyway without version number? My gut feeling is no; as we only want to print info about files that are definitely owned by systemd.

Comment (arianvp) [reply src/boot/bootctl-status.c:249]: I'm not sure what the intended behaviour is. As before #27069 the printing would just break if a file not owned by systemd was found

**Comment by poettering** [comment]
uff. the code so far is really broken.

enumerate_binaries() is invoked on two paths: the systemd-specific /EFI/systemd/ one, where everything contained should be owned by us. And the generic EFI/BOOT/boot* which might be owned by anyone.

For the latter we definitely should generate output, because it very likely points to windows or so.

hence, the current code is broken if it just skips over files that otherwise match but carry no recognizable version info. We should display the files without version info. i.e. it's highly relevant to know that there is some EFI program installed as EFI/BOOT/boot*.efi handler, even if it's now ours.

/cc @keszybz

**Comment by poettering** [comment]
(get_file_version() logging is all over the place now, it logs about some errors, but not about others, ESRCH seems to be special, other errors are not).

**Comment by arianvp** [comment]
So sounds like I should remove the `continue` that skips over unknown files and instead print without version number if `ESRCH` is returned.

File: src/boot/bootctl-status.c [hunk 2]
```diff
@@ -243,10 +241,10 @@
                 /* Do not output this entry immediately, but store what should be printed in a state
                  * variable, because we only will know the tree glyph to print (branch or final edge) once we
                  * read one more entry */
-                if (r > 0)
-                        r = asprintf(previous, "/%s/%s (%s%s%s)", path, de->d_name, ansi_highlight(), v, ansi_normal());
-                else
+                if (r == -ESRCH) /* No systemd-owned file but still interesting to print */
```
[the remark this answers is not in the document]
Comment (arianvp) [reply src/boot/bootctl-status.c:245]: Thanks CodeQL ! that was useful!

**Comment by keszybz** [comment]
LGTM.

**Pull request merged**